### DISCUSSION TITLE
Netbox cable splitting functionality requires input and output to match

### DISCUSSION THREAD

**Discussion by MajesticFalcon** [netbox-community/netbox#12372] [Q&A] [answered]
I've spent several hours trying to figure out how to model a typical next gen PON network using all the new cable features that came out.

<img width="1051" alt="image" src="https://user-images.githubusercontent.com/8593209/234959921-c33a9e58-fde4-4d96-ab76-573342e55926.png">

I've played with positions, multi-interface cables, etc. I can't seem to get it. 

I am stuck on how to show a cable trace where an interface splits into two of the paths.

Below is the closest I have gotten. I need the one front port to map to two rear ports.

<img width="403" alt="image" src="https://user-images.githubusercontent.com/8593209/234960637-7a3d1113-fe0c-4b51-bcb1-29ba28cd0e2b.png">

**Comment by MajesticFalcon** [comment]
It seems netbox will only do the multi output if the multi-input exists.

This fixes the bottom but the top is wrong. 
<img width="460" alt="Screenshot 2023-04-27 at 2 29 13 PM" src="https://user-images.githubusercontent.com/8593209/234972954-87aec5d3-ba38-46d4-8375-539735ef1524.png">

**Comment by candlerb** [comment, marked as answer]
There are quite a few limitations with fibre modelling in Netbox, and being unable to model PON splitters is one of them.

The multi-position stuff (i.e. 1 rear port with N positions <-> N front ports) is for modelling a multi-strand trunk cable, like an MPO, used to join two patch panels together - not for PON splitting. A single "N-strand" (implied) rear cable breaks out into N front ports. Netbox is able to trace such a cable from the Nth front port on one patch panel, through the trunk cable, to the Nth front port on the other patch panel - the rear port "position" is implicitly the strand offset within the trunk cable. The cable is still just a single cable, and doesn't actually contain explicitly modelled "strands".

Netbox also doesn't model SFPs, and it doesn't model separate transmit/receive fibre strands. So every "interface" can only accept a single "cable".  However, you can have a cable which connects to an interface at one end, and two ODF ports at the other end. It's still considered a single "connection".

Similarly, if you have a QFP port at one end of a cable, and four distinct SFP+ ports at the other end, you can't treat these as separate connection paths (even if the QFP port is configurable as 4 separate interfaces).

I'm afraid this info doesn't help you. The best you can do for now, I believe, is to model all the passive ports as "interfaces" - as if they were switches - but you won't be able to trace cables through them, you'll have to trace in hops.
